src/Collector/ComponentCollector.php

```diff
@@ -51,6 +66,27 @@ public function getNodeType(): string
      */
     public function processNode(Node $node, Scope $scope): ?array
     {
+        // Use this and process stmts in abstract collector
+        if ($node instanceof TraitUse) {
+            $traitComponents = [];
+            foreach ($node->traits as $trait) {
+                $traitName = $this->nameResolver->resolve($trait);
+                if ($traitName === null) {
+                    continue;
+                }
+
+                $traitClassReflection = $this->reflectionProvider->getClass($traitName);
+                $traitFileName = $traitClassReflection->getFileName();
+                if ($traitFileName === null) {
+                    continue;
+                }
+
+                $traitNodes = $this->parser->parseFile($traitFileName);
+                $traitComponents = array_merge($traitComponents, $this->processStmts($traitNodes, $scope));
```

Comment:
Are you sure this will not corrupt Scope with unvanted changes? Mybe clone Scope before passing it?

Reply:
scope changed